File: contrib/pyre/packages/journal/Console.py
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis
# orthologue
# (c) 1998-2019 all rights reserved
#


# packages
import pyre


# the implemented interfaces
from .Device import Device


# declaration
class Console(pyre.component, family="journal.devices.console", implements=Device):
    """
    A device that sends journal messages to the standard output
    """


    # public state
    renderer = Device.Renderer()
    renderer.doc = "the formatting strategy for journal entries"


    # interface
    @pyre.export
    def record(self, page, metadata):
        """
        Record a journal entry
        """
        # assemble the text
        text = '\n'.join(self.renderer.render(page, metadata))
        # print it to stdout
        print(text, flush=True)
        # and return
        return self


# end of file

File: extern/pyre/lib/journal/Console.cc
// -*- C++ -*-
//
// michael a.g. aïvázis
// orthologue
// (c) 1998-2019 all rights reserved
//


// for the build system
#include <portinfo>

// packages
#include <map>
#include <vector>
#include <string>
#include <iostream>

// local types
#include "Renderer.h"
#include "Device.h"
#include "Streaming.h"
#include "Console.h"


// interface
// record a diagnostic
void
pyre::journal::Console::
record(entry_t & entry, metadata_t & metadata)
{
    // get the renderer to convert the diagnostic parts into a string
    string_t text = _renderer->render(entry, metadata);
    // print it out
    std::cout << text;
    // and return
    return;
}


// destructor
pyre::journal::Console::
~Console()
{
    delete _renderer;
}


// the default constructor
pyre::journal::Console::
Console() :
    pyre::journal::Streaming(std::cout),
    _renderer(new Renderer())
{}


// end of file
